diff --git a/QuoteGeneration/pccs/pcs_client/pcs_client.js b/QuoteGeneration/pccs/pcs_client/pcs_client.js
index cf675a8..8872676 100644
--- a/QuoteGeneration/pccs/pcs_client/pcs_client.js
+++ b/QuoteGeneration/pccs/pcs_client/pcs_client.js
@@ -66,7 +66,7 @@ async function do_request(url, options) {
       if (!options.headers) {
         options.headers = {};
       }
-      options.headers['Ocp-Apim-Subscription-Key'] = Config.get('ApiKey');
+      const apikey = Config.get('ApiKey'); if (apikey) options.headers['Ocp-Apim-Subscription-Key'] = apikey;
     }
 
     // global opitons ( proxy, timeout, etc)
@@ -128,9 +128,11 @@ export async function getCerts(enc_ppid, pceid) {
       pceid: pceid,
     },
     method: 'GET',
-    headers: { 'Ocp-Apim-Subscription-Key': Config.get('ApiKey') },
+    headers: {},
   };
 
+  const apikey = Config.get('ApiKey'); if (apikey) options.headers['Ocp-Apim-Subscription-Key'] = apikey;
+
   return do_request(Config.get('uri') + 'pckcerts', options);
 }
 
@@ -142,11 +144,12 @@ export async function getCertsWithManifest(platform_manifest, pceid) {
     },
     method: 'POST',
     headers: {
-      'Ocp-Apim-Subscription-Key': Config.get('ApiKey'),
       'Content-Type': 'application/json',
     },
   };
 
+  const apikey = Config.get('ApiKey'); if (apikey) options.headers['Ocp-Apim-Subscription-Key'] = apikey;
+
   return do_request(Config.get('uri') + 'pckcerts', options);
 }
 
